src/main/java/io/openmessaging/storage/dledger/DLedgerEntryPusher.java

```diff
@@ -796,6 +796,7 @@ public EntryHandler(Logger logger) {
                     future.complete(buildResponse(request, DLedgerResponseCode.UNEXPECTED_ARGUMENT.getCode()));
                     break;
             }
+            wakeup();
```

Comment:
I don’t understand why wakeup called here, the thread  is already working.

Reply:
long nextIndex = dLedgerStore.getLedgerEndIndex() + 1;
                    Pair<PushEntryRequest, CompletableFuture<PushEntryResponse>> pair = writeRequestMap.remove(nextIndex);
                    if (pair == null) {
                        checkAbnormalFuture(dLedgerStore.getLedgerEndIndex());
                        waitForRunning(1);
                        return;
                    }
                    PushEntryRequest request = pair.getKey();
                    if (dLedgerConfig.isEnableBatchPush()) {
                        handleDoBatchAppend(nextIndex, request, pair.getValue());
                    } else {
                        handleDoAppend(nextIndex, request, pair.getValue());
                    }

src/main/java/io/openmessaging/storage/dledger/DLedgerEntryPusher #1031 line

when has no new message ,writeRequestMap is empty，but when  new message arrive，put in writeRequestMap . It didn't wake up the thread

Reply:
> ```
>        long nextIndex = dLedgerStore.getLedgerEndIndex() + 1;
>                 Pair<PushEntryRequest, CompletableFuture<PushEntryResponse>> pair = writeRequestMap.remove(nextIndex);
>                 if (pair == null) {
>                     checkAbnormalFuture(dLedgerStore.getLedgerEndIndex());
>                     waitForRunning(1);
>                     return;
>                 }
>                 PushEntryRequest request = pair.getKey();
>                 if (dLedgerConfig.isEnableBatchPush()) {
>                     handleDoBatchAppend(nextIndex, request, pair.getValue());
>                 } else {
>                     handleDoAppend(nextIndex, request, pair.getValue());
>                 }
> ```
> 
> src/main/java/io/openmessaging/storage/dledger/DLedgerEntryPusher #1031 line
> 
> when has no new message ,writeRequestMap is empty，but when new message arrive，put in writeRequestMap . It didn't wake up the thread

ok, I understand